### PR TITLE
[NOT TO BE MERGED] Just some example project

### PR DESCRIPTION
**Pull request by vvs-**
This is NOT intended for merging. It's just a quick hack and a starting point for discussion. Also it could be useful for those adventurers who want to quickly try the new code on Linux.

The proposed build system should be more complete than the old system was. The best starting point should be @stohrendorf build system based on Hunter.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,12 +1,176 @@
...
+#  make
 
-cmake_minimum_required(VERSION 2.8)
+CMAKE_MINIMUM_REQUIRED(VERSION 2.8)
 
-project(OpenTomb)
+PROJECT(OpenTomb)
 
-message(WARNING "This CMakeLists.txt does not generate executables, it is only used to process the documentation.")
+SET(CMAKE_CXX_FLAGS "-std=c++0x")
```
Comment (stohrendorf) [inline CMakeLists.txt]: Maybe bump that up to `c++11` or even `c++14`.

**Comment by stohrendorf** [comment]
The downside of hunter is that it requires an external file to add new packages that aren't already provided by hunter. Also, I have a few hacked (forked) repos of popular libs to make them compatible with hunter. That's maybe not such a good idea.

**Comment by vvs-** [comment]
The OpenTomb established workflow depends on local in tree libraries. So, it's not that bad in comparison. Also, you already used a macro that can select between system and Hunter libraries. It should work well in practice. We should test it on OS X and Windows to assess the consequences.

**Comment by stohrendorf** [comment]
From what I can tell it both works on Windows and Linux, but I can't say anything about OS X. Especially boost (if used) takes hours (!) to build on Windows, and I usually prefer to download a pre-built boost dev package and force the project to use that.

**Pull request closed without merging**